presto-tpch/src/main/java/com/facebook/presto/tpch/TpchRecordSet.java

```diff
@@ -197,26 +204,36 @@ public void close()
             closed = true;
         }
 
-        private Map<ColumnHandle, NullableValue> rowMap()
+        private boolean rowMatchesPredicate()
         {
-            ImmutableMap.Builder<ColumnHandle, NullableValue> builder = ImmutableMap.builder();
-            for (int field = 0; field < columnHandles.size(); ++field) {
-                Type type = columnTypes.get(field);
-                builder.put(columnHandles.get(field), NullableValue.of(type, getPrestoObject(field, type)));
+            if (predicate.isAll()) {
+                return true;
             }
-            return builder.build();
+            Map<ColumnHandle, NullableValue> rowMap = predicate.getDomains().get().keySet().stream()
+                    .collect(toImmutableMap(
+                            column -> column,
+                            column -> {
+                                TpchColumnHandle tpchColumnHandle = (TpchColumnHandle) column;
+                                Type type = tpchColumnHandle.getType();
+                                TpchColumn tpchColumn = table.getColumn(tpchColumnHandle.getColumnName());
+                                return NullableValue.of(type, getPrestoObject(tpchColumn, type));
+                            }));
+
+            TupleDomain<ColumnHandle> rowTupleDomain = TupleDomain.fromFixedValues(rowMap);
+
+            return predicate.contains(rowTupleDomain);
```

Comment:
I think this will be always `false` when `predicate` doesn't contain all table columns. Please validate.

Reply:
Ok if you say so.

However, `TupleDomain#contains` does `columnWiseUnion` which shouldn't be equal to `predicate` if `rowTupleDomain` contains more columns than `predicate`. I remember @findepi hit the same issue

Maybe add test to `AbstractTestQueries`:
```
select * from orders where orderstatus='F'
```
?

Reply:
> if rowTupleDomain contains more columns than predicate

`rowTupleDomain` will have exactly the same number of columns as `predicate`, as it is build from `predicate`.

`AbstractTestQueries` already contains queries containing: `SELECT .. FROM orders WHERE orderstatus = 'F' `